core/src/main/scala/kafka/server/NoOpObserver.scala

```diff
@@ -43,4 +43,9 @@ class NoOpObserver extends Observer {
     */
   def close(timeout: Long, unit: TimeUnit): Unit = {}
 
+  /**
+   * Check client library.
+   */
+  def checkClientLibrary(hasNonXinfraClient: Boolean, clientId: String): Unit = {}
```

Comment:
name should better be isXinfraClient? basically it's 1-1 mapping between the boolean and the clientId, i.e. the clientId is either a xinfra client or a kafka client

Reply:
This name is what we're using in the plumbing with bean counter. Basically we can't add a new field for different client types without exploding the amount of metrics that bean-counter is producing so instead we're adding a flag for whether this topic hasNonXinfraClients or not. There's discussions on how we can add more detailed information in another pass without blowing up the amount of metrics but for now we're sticking with whether a topic hasNonXinfraClient accessing it

Reply:
what I mean is, instead of hasNonXinfraClients, the naming should be isNonXinfra or isXinfra, since a client Id either belongs to a LKC client or a XC client.

Reply:
updated